third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -205,13 +216,15 @@ index 1240623b5..813b26ec8 100755
  
      # Tokenizer check: this script requires a fast tokenizer.
      if not isinstance(tokenizer, PreTrainedTokenizerFast):
-@@ -599,6 +596,41 @@ def main():
+@@ -599,6 +598,43 @@ def main():
      def compute_metrics(p: EvalPrediction):
          return metric.compute(predictions=p.predictions, references=p.label_ids)
  
 +    nncf_config = None
 +    if training_args.nncf_config is not None:
 +        nncf_config = NNCFConfig.from_json(training_args.nncf_config)
++        nncf_config.register_extra_structs([BNASLRSchedulerArgs(nncf_config)])
++        update_bnas_lr_scheduler_kwargs(nncf_config, training_args)
```

Comment:
In each run_{task}.py, I register extra struct for LR parameters and update the "training_args". I didn't put it in trainer.py mainly because nncf_config is not passed to class Trainer.